File: src/main/java/com/love/java/Box.java
package com.love.java;


public abstract class Box {
    float length;
    float width;
    float height;

    public boolean validate(float length, float width, float height) {
        if (length <= this.length && width <= this.width && height <= this.height) {
            return true;
        } else {
            return false;
        }
    }
    public abstract void useBox();
}

File: src/main/kotlin/com/love/kotlin/Box.kt
package com.love.kotlin

abstract class Box {
    open val length: Float = 0.0f
    open val width: Float = 0.0f
    open val height: Float = 0.0f
    fun validate(length: Float, width: Float, height: Float): Boolean {
        return length <= this.length && width <= this.width && height <= this.height
    }
    abstract fun useBox()
}